posts/2022-08-04-jakarta-core-profile-22009-beta.adoc

```diff
@@ -0,0 +1,276 @@
+If set to `false`, then archives that do not contain a `beans.xml` file are not scanned for annotated classes.
+
+The `emptyBeansXmlCDI3Compatibility` attribute applies only to CDI 4.0.
+If set to `true`, then an archive that contains an empty `beans.xml` file is treated as an Explicit Bean Archive, as it was in CDI 3.0 and earlier.
+
+If set to `false`, which is the default, then an archive that contains an empty `beans.xml` file is treated as an Implicit Bean Archive. This is equivalent to setting `bean-discovery-mode="annotated"`.
+
+[source, xml]
+----
+<featureManager>
+  <feature>cdi-4.0</feature>
+</featureManager>
+
+<cdi enableImplicitBeanArchives="false" emptyBeansXmlCDI3Compatibility="true"/>
+----
+
+[#expression]
+== Jakarta EE 10 Expression Language 5.0
+
+Jakarta EE 10 Expression Language 5.0 includes a number of new features, deprecations, and specification clarifications.  One of the new features is the addition of the link:https://jakarta.ee/specifications/expression-language/5.0/apidocs/jakarta.el/jakarta/el/methodreference[MethodReference] which "Provides information about the method to which a method expression resolves." This new class allows developers to access any annotations present on a particular method using the `getAnnotations()` method.
+
+To enable Expression Language 5.0, add the `expressionLanguage-5.0` feature to the list of features in your server.xml file:
+[source,xml]
+----
+   <featureManager>
+        <feature>expressionLanguage-5.0</feature>
+    </featureManager>
+----
+
+More information about Expression Language 5.0 can be found in: 
+
+* link:https://jakarta.ee/specifications/expression-language/5.0/jakarta-expression-language-spec-5.0.html#changes-between-5-0-and-4-0[Changes between the Expression Language 4.0 and Expression Language 5.0]
+
+* link:https://jakarta.ee/specifications/expression-language/5.0/[Expression Language 5.0 specification]
+
+* link:{url-prefix}/docs/latest/reference/feature/expressionLanguage.html[Open Liberty Expression Language feature documentation]
+
+[#microprofile]
+== MicroProfile OpenAPI 3.1 
+
+OpenAPI is a standardised way of documenting REST APIs in a JSON or YAML format. It helps you generate and serve OpenAPI documentation for your REST applications that are built by using JAX-RS or Jakarta restfulWS. This can be useful for developers to test out the API during development, or for people using the API in production.
+
+To use MicroProfile OpenAPI 3.1, add the `mpOpenAPI-3.1` feature to your server.xml:
+[source,xml]
+----
+   <featureManager>
+        <feature>mpOpenAPI-3.1</feature>
+    </featureManager>
+----
+
+ Next, deploy your rest application and then visit the `/openapi` endpoint to view the OpenAPI documentation, or visit `/openapi/ui` for a user interface where you can browse through it. You can enhance the documentation by using annotations to add things like textual descriptions to each method. The annotations are available as a maven dependency:
```

Comment:
```suggestion
Next, deploy your rest application and then visit the `/openapi` endpoint to view the OpenAPI documentation, or visit `/openapi/ui` for a user interface where you can browse through it. You can enhance the documentation by using annotations to add things like textual descriptions to each method. The annotations are available as a maven dependency:
```

Reply:
This extra space appears to cause the markdown to do weird things at least in the preview.